pkg/provisioner/ironic/testserver/ironic.go

```diff
@@ -75,3 +87,45 @@ func (m *IronicMock) WithNodeStatesProvision(nodeUUID string) *IronicMock {
 	m.ResponseWithCode("/v1/nodes/"+nodeUUID+"/states/provision", "{}", http.StatusAccepted)
 	return m
 }
+
+// NoNode configures the server so /v1/nodes/name returns a 404
+func (m *IronicMock) NoNode(name string) *IronicMock {
+	m.NotFound(fmt.Sprintf("/v1/nodes/%s", name))
+	return m
+}
+
+// CreateNodes configures the server so POSTing to /v1/nodes saves the data
+func (m *IronicMock) CreateNodes() *IronicMock {
+	m.Handler("/v1/nodes", func(w http.ResponseWriter, r *http.Request) {
+		bodyRaw, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			m.logRequest(r, fmt.Sprintf("ERROR: %s", err))
+			http.Error(w, fmt.Sprintf("%s", err), 500)
```

Comment:
Minor point, `http.StatusInternalServerError` could be more readable

Reply:
Fixed